tools/ramdump/HowToUseRamdump.md

```diff
@@ -180,3 +207,70 @@ Call Trace of Crashed Task :[appmain] with pid :2 and state :TSTATE_TASK_RUNNING
 [<40c9fec>] task_start+0x64         [Line 173 of \"task/task_start.c\"]
 ********************************************************************
 ```
+## How to port RAMDUMP
+To port ramdump for a new board, do the following steps:
+
+1. Add low level chip specific API's to receive and transfer characters through UART:
+a. **up_putc()** : Output one byte on the serial console
+```
+ * Prototype: int up_putc(int ch)
+ *   Output one byte on the serial console
+ * Input Parameters:
+ *   ch - chatacter to output
+ * Returned Value:
+ *   sent character
+```
+b. **up_getc()** : Read one byte from the serial console
+```
+ * Prototype: int up_getc(void)
+ *   Read one byte from the serial console
+ * Input Parameters:
+ *   none
+ * Returned Value:
+ *   int value, -1 if error, 0~255 if byte successfully read
+```
+c. **up_puts()** : Output one byte on the serial console
+```
+ * Prototype: void up_puts(const char *str)
+ *   Output one byte on the serial console
+ * Input Parameters:
+ *   ch - chatacter to output
+ * Returned Value:
+ *   none
+```
+
+2. Source the low level API file in the Make.defs of os/arch/<cpu_name>/src/<chip_name>
+```
+CMN_CSRCS += <chip_name>_serial.c
+```
+3. Source the <chip_name>_crashdump.c file in chip and board Makefile (os/board/<board_name>/src)
+```
+DEPPATH += --dep-path $(TOPDIR)/board/common
+VPATH += :$(TOPDIR)/board/common
+
+ifeq ($(CONFIG_BOARD_CRASHDUMP),y)
+CSRCS += <chip_name>_crashdump.c
```

Comment:
The crashdump.c is in common folder so that `<chip_name>_crashdump.c` is wrong.
```
DEPPATH += --dep-path $(TOPDIR)/board/common
VPATH += :$(TOPDIR)/board/common
ifeq ($(CONFIG_BOARD_CRASHDUMP),y)
CSRCS += crashdump.c
endif
```